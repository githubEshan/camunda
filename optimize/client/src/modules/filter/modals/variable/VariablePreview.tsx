/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
 * one or more contributor license agreements. See the NOTICE file distributed
 * with this work for additional information regarding copyright ownership.
 * Licensed under the Camunda License 1.0. You may not use this file
 * except in compliance with the Camunda License 1.0.
 */

import {ReactNode} from 'react';
import {Tag} from '@carbon/react';

import {t} from 'translation';

interface VariablePreviewProps {
  variableName: string;
  filter: {
    operator?: string;
    values: (string | number | boolean | null)[];
  };
  type?: string;
}

export default function VariablePreview({
  variableName,
  filter: {operator, values},
  type = 'variable',
}: VariablePreviewProps) {
  const createOperator = (val: ReactNode) => <span> {val} </span>;

  const operatorText = (
    <span>
      {operator === 'not in' || operator === 'not contains'
        ? createOperator(t('common.filter.list.operators.nor'))
        : createOperator(t('common.filter.list.operators.or'))}
    </span>
  );

  return (
    <>
      <span title={`${t('report.table.rawData.' + type)}: ${variableName}`}>
        <Tag type="blue" className="parameterName">
          {t('report.table.rawData.' + type)}: <b>{variableName}</b>
        </Tag>
      </span>
      <span className="filterText">
        {(!operator || operator === 'in' || operator === '=') &&
          createOperator(t('common.filter.list.operators.is'))}
        {operator === 'contains' && createOperator(t('common.filter.list.operators.contains'))}
        {operator === 'not contains' &&
          createOperator(t('common.filter.list.operators.notContains'))}
        {operator === 'not in' &&
          (values.length === 1
            ? createOperator(t('common.filter.list.operators.not'))
            : createOperator(t('common.filter.list.operators.neither')))}
        {operator === '<' && createOperator(t('common.filter.list.operators.less'))}
        {operator === '>' && createOperator(t('common.filter.list.operators.greater'))}
        {values.map((value, idx) => (
          <span key={idx}>
            {value === null ? (
              <>
                <b>{t('common.null')}</b>
                {operatorText}
                <b>{t('common.undefined')}</b>
              </>
            ) : (
              <b>{value.toString()}</b>
            )}
            {idx < values.length - 1 && operatorText}
          </span>
        ))}
      </span>
    </>
  );
}
