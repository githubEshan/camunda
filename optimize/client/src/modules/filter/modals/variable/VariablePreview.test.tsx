/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
 * one or more contributor license agreements. See the NOTICE file distributed
 * with this work for additional information regarding copyright ownership.
 * Licensed under the Camunda License 1.0. You may not use this file
 * except in compliance with the Camunda License 1.0.
 */

import React from 'react';
import {shallow} from 'enzyme';

import VariablePreview from './VariablePreview';

it('should combine multiple variable values with or', () => {
  const node = shallow(
    <VariablePreview
      filter={{operator: 'in', values: ['varValue', 'varValue2']}}
      variableName="varName"
    />
  );

  expect(node).toMatchSnapshot();
});

it('should combine multiple variable values with or', () => {
  const node = shallow(
    <VariablePreview
      filter={{operator: 'not in', values: [null, 'varValue', 'varValue2']}}
      variableName="varName"
    />
  );

  expect(node).toMatchSnapshot();
});

it('should use less/greater for comparison operators', () => {
  const node = shallow(
    <VariablePreview filter={{operator: '<', values: ['varValue']}} variableName="varName" />
  );

  expect(node).toMatchSnapshot();
});

it('should use contains operators', () => {
  const node = shallow(
    <VariablePreview filter={{operator: 'contains', values: ['varValue']}} variableName="varName" />
  );

  expect(node).toMatchSnapshot();
});

it('should display correct preview even if no operator is defined', () => {
  const node = shallow(<VariablePreview filter={{values: [true, false]}} variableName="varName" />);

  expect(node).toMatchSnapshot();
});
