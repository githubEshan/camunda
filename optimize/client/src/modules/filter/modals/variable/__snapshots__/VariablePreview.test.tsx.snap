// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should combine multiple variable values with or 1`] = `
<Fragment>
  <span
    title="Var: varName"
  >
    <ForwardRef(Tag)
      className="parameterName"
      type="blue"
    >
      Var
      : 
      <b>
        varName
      </b>
    </ForwardRef(Tag)>
  </span>
  <span
    className="filterText"
  >
    <span>
       
      is
       
    </span>
    <span
      key="0"
    >
      <b>
        varValue
      </b>
      <span>
        <span>
           
          or
           
        </span>
      </span>
    </span>
    <span
      key="1"
    >
      <b>
        varValue2
      </b>
    </span>
  </span>
</Fragment>
`;

exports[`should combine multiple variable values with or 2`] = `
<Fragment>
  <span
    title="Var: varName"
  >
    <ForwardRef(Tag)
      className="parameterName"
      type="blue"
    >
      Var
      : 
      <b>
        varName
      </b>
    </ForwardRef(Tag)>
  </span>
  <span
    className="filterText"
  >
    <span>
       
      is neither
       
    </span>
    <span
      key="0"
    >
      <b>
        null
      </b>
      <span>
        <span>
           
          nor
           
        </span>
      </span>
      <b>
        undefined
      </b>
      <span>
        <span>
           
          nor
           
        </span>
      </span>
    </span>
    <span
      key="1"
    >
      <b>
        varValue
      </b>
      <span>
        <span>
           
          nor
           
        </span>
      </span>
    </span>
    <span
      key="2"
    >
      <b>
        varValue2
      </b>
    </span>
  </span>
</Fragment>
`;

exports[`should display correct preview even if no operator is defined 1`] = `
<Fragment>
  <span
    title="Var: varName"
  >
    <ForwardRef(Tag)
      className="parameterName"
      type="blue"
    >
      Var
      : 
      <b>
        varName
      </b>
    </ForwardRef(Tag)>
  </span>
  <span
    className="filterText"
  >
    <span>
       
      is
       
    </span>
    <span
      key="0"
    >
      <b>
        true
      </b>
      <span>
        <span>
           
          or
           
        </span>
      </span>
    </span>
    <span
      key="1"
    >
      <b>
        false
      </b>
    </span>
  </span>
</Fragment>
`;

exports[`should use contains operators 1`] = `
<Fragment>
  <span
    title="Var: varName"
  >
    <ForwardRef(Tag)
      className="parameterName"
      type="blue"
    >
      Var
      : 
      <b>
        varName
      </b>
    </ForwardRef(Tag)>
  </span>
  <span
    className="filterText"
  >
    <span>
       
      contains
       
    </span>
    <span
      key="0"
    >
      <b>
        varValue
      </b>
    </span>
  </span>
</Fragment>
`;

exports[`should use less/greater for comparison operators 1`] = `
<Fragment>
  <span
    title="Var: varName"
  >
    <ForwardRef(Tag)
      className="parameterName"
      type="blue"
    >
      Var
      : 
      <b>
        varName
      </b>
    </ForwardRef(Tag)>
  </span>
  <span
    className="filterText"
  >
    <span>
       
      is less than
       
    </span>
    <span
      key="0"
    >
      <b>
        varValue
      </b>
    </span>
  </span>
</Fragment>
`;
